src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1878,22 +1876,12 @@ WorkerAggregateExpressionList(Aggref *originalAggregate,
 		walkerContext->repartitionSubquery)
 	{
 		Aggref *aggregate = (Aggref *) copyObject(originalAggregate);
-		List *aggTargetEntryList = aggregate->args;
-		TargetEntry *distinctTargetEntry = (TargetEntry *) linitial(aggTargetEntryList);
-		List *columnList = pull_var_clause_default((Node *) distinctTargetEntry);
+		List *columnList = pull_var_clause_default((Node *) aggregate);
 		ListCell *columnCell = NULL;
-		List *processedColumnList = NIL;
-
 		foreach(columnCell, columnList)
 		{
 			Var *column = (Var *) lfirst(columnCell);
-			if (list_member(processedColumnList, column))
-			{
-				continue;
-			}
-
-			processedColumnList = lappend(processedColumnList, column);
-			workerAggregateList = lappend(workerAggregateList, copyObject(column));
+			workerAggregateList = list_append_unique(workerAggregateList, column);
```

Comment:
We are doing very similar thing, but instead of the aggregate itself we are adding the columns referenced from that.
I do not see any additional benefit by adding another level of copy here.